src/Features/CSharpTest/ImplementAbstractClass/ImplementAbstractClassTests.cs

```diff
@@ -2471,5 +2471,45 @@ protected override void M()
                 }
                 """);
         }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/70530")]
+        public async Task TestNullableGenericConstraintToSpecificReferenceType()
+        {
+            await TestAllOptionsOffAsync(
+                """
+                #nullable enable
+                interface I<out T> { }
+
+                class C { }
+
+                abstract class Problem
+                {
+                    protected abstract void M<T>(I<T?> i) where T : C;
+                }
+
+                class [|Bad|] : Problem
+                {
+                }
+                """,
+                """
+                #nullable enable
+                interface I<out T> { }
+
+                class C { }
+
+                abstract class Problem
+                {
+                    protected abstract void M<T>(I<T?> i) where T : C;
```

Comment:
can we have a test where instead of T being dependent on a class C, it is dependent on *another* type parameter (which itself has its own constraints)?

basically, i think we need to be able to look atrbitrarily deep (unless the compiler reifies the constraint information all the way through the relationship hierarchy).

Reply:
> unless the compiler reifies the constraint information all the way through the relationship hierarchy

Seems to be the case, I changed the test as you suggested and it still passes